src/ft/mod.rs

```diff
@@ -173,7 +173,113 @@ impl Rasterize for FreeTypeRasterizer {
+        }
+
+        unsafe {
+            let ft_lib = self.loader.library.raw();
+            freetype::ffi::FT_Library_SetLcdFilter(ft_lib, face.lcd_filter);
+        }
+
+        face.ft_face.load_glyph(index, face.load_flags)?;
+
+        let glyph = face.ft_face.glyph();
+
+        // Generate synthetic bold.
+        if face.embolden {
+            unsafe {
+                freetype_sys::FT_GlyphSlot_Embolden(glyph.raw()
+                    as *const freetype_sys::FT_GlyphSlotRec
+                    as *mut freetype_sys::FT_GlyphSlotRec);
+            }
+        }
+
+        let advance = unsafe {
+            // Transform glyphs with the matrix from Fontconfig. Primarily used to generate italics.
+            let raw_glyph = face.ft_face.raw().glyph;
+            if let Some(matrix) = face.matrix.as_ref() {
+                // Check that the glyph is a vectorial outline, not a bitmap.
+                if (*raw_glyph).format == freetype_sys::FT_GLYPH_FORMAT_OUTLINE {
+                    let outline = &(*raw_glyph).outline;
+
+                    freetype_sys::FT_Outline_Transform(outline, matrix);
+                }
+            }
+
+            // Don't render bitmap glyphs, it results in error with freestype 2.11.0.
+            if (*raw_glyph).format != freetype_sys::FT_GLYPH_FORMAT_BITMAP {
+                glyph.render_glyph(face.render_mode)?;
+            }
+
+            let advance = (*raw_glyph).advance;
+            (advance.x as i32 / 64, advance.y as i32 / 64)
+        };
+
+        let (pixel_height, pixel_width, buffer) =
+            Self::normalize_buffer(&glyph.bitmap(), &face.rgba)?;
+
+        let mut rasterized_glyph = RasterizedGlyph {
+            character: glyph_key.character,
+            top: glyph.bitmap_top(),
+            left: glyph.bitmap_left(),
+            width: pixel_width,
+            height: pixel_height,
+            advance,
+            buffer,
+        };
+
+        if index == MISSING_GLYPH_INDEX {
+            Err(Error::MissingGlyph(rasterized_glyph))
+        } else {
+            if face.colored {
+                let fixup_factor = match face.pixelsize_fixup_factor {
+                    Some(fixup_factor) => fixup_factor,
+                    None => {
+                        // Fallback if the user has bitmap scaling disabled.
+                        let metrics = face.ft_face.size_metrics().ok_or(Error::MetricsNotFound)?;
+                        f64::from(pixelsize) / f64::from(metrics.y_ppem)
+                    },
+                };
+
+                // Scale glyph advance.
+                rasterized_glyph.advance.0 = (advance.0 as f64 * fixup_factor).round() as i32;
+                rasterized_glyph.advance.1 = (advance.1 as f64 * fixup_factor).round() as i32;
+
+                rasterized_glyph = downsample_bitmap(rasterized_glyph, fixup_factor);
+            }
+
+            Ok(rasterized_glyph)
+        }
```

Comment:
Would you mind refactoring it into a match while we're at it? This nested isn't that readable tbh. If you don't want to, we can go ahead.

Reply:
What exactly do you want to refactor into a match?

I'd suggest to just early return in missing glyphs to reduce nesting?

Reply:
I was thinking about something like
```rust
match index {
    MISSING_GLYPH_INDEX => Err(Error::MissingGlyph(rasterized_glyph)),
    _ if face.colored => {
        let fixup_factor = match face.pixelsize_fixup_factor {
            Some(fixup_factor) => fixup_factor,
            None => {
                // Fallback if the user has bitmap scaling disabled.
                let metrics = face.ft_face.size_metrics().ok_or(Error::MetricsNotFound)?;
                f64::from(pixelsize) / f64::from(metrics.y_ppem)
            }
        };

        // Scale glyph advance.
        rasterized_glyph.advance.0 = (advance.0 as f64 * fixup_factor).round() as i32;
        rasterized_glyph.advance.1 = (advance.1 as f64 * fixup_factor).round() as i32;

        rasterized_glyph = downsample_bitmap(rasterized_glyph, fixup_factor);
        Ok(rasterized_glyph)
    }
    _ => Ok(rasterized_glyph),
}
```

But early returns are fine too, I think.